core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -135,22 +165,22 @@ func (l *logPollerWrapper) LatestEvents() ([]evmRelayTypes.OracleRequest, []evmR
 		l.lggr.Debug("LatestEvents: no non-zero coordinators to check")
 		return resultsReq, resultsResp, errors.New("no non-zero coordinators to check")
 	}
-	if latest < nextBlock {
-		l.lggr.Debugw("LatestEvents: no new blocks to check", "latest", latest, "nextBlock", nextBlock)
-		return resultsReq, resultsResp, nil
-	}
 
 	for _, coordinator := range coordinators {
-		requestLogs, err := l.logPoller.Logs(nextBlock, latest, functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic(), coordinator)
+		endBlock := latest - l.requestBlockOffset
```

Comment:
Add a sanity check in the constructor to make sure *blockOffset is not larger than pastBlocksToPoll